packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,323 @@
+        }
+
+        Safe targetSafe = Safe(payable(_safe));
+
+        // Get current owners
+        address[] memory owners = targetSafe.getOwners();
+
+        // Remove all owners after the first one
+        // Note: This loop is safe as real-world Safes have limited owners (typically < 10)
+        // Gas limits would only be a concern with hundreds/thousands of owners
+        while (owners.length > 1) {
+            targetSafe.execTransactionFromModule({
+                to: _safe,
+                value: 0,
+                operation: Enum.Operation.Call,
+                data: abi.encodeCall(OwnerManager.removeOwner, (SENTINEL_OWNER, owners[0], 1))
+            });
+            owners = targetSafe.getOwners();
+        }
+
+        // Now swap the remaining single owner with the fallback owner
+        targetSafe.execTransactionFromModule({
+            to: _safe,
+            value: 0,
+            operation: Enum.Operation.Call,
+            data: abi.encodeCall(OwnerManager.swapOwner, (SENTINEL_OWNER, owners[0], safeConfigs[_safe].fallbackOwner))
+        });
+
+        // Sanity check: verify the fallback owner is now the only owner
+        address[] memory finalOwners = targetSafe.getOwners();
+        if (finalOwners.length != 1 || finalOwners[0] != safeConfigs[_safe].fallbackOwner) {
+            revert LivenessModule2_OwnershipTransferFailed();
+        }
+
+        // Reset the challenge state to allow a new challenge
+        delete challengeStartTime[_safe];
+
+        emit ChallengeSucceeded(_safe, safeConfigs[_safe].fallbackOwner);
+    }
+
+    /// @notice Asserts that the module is configured for the given Safe.
+    /// @param _safe The Safe address to check.
+    function _assertModuleConfigured(address _safe) internal view {
+        ModuleConfig storage config = safeConfigs[_safe];
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+    }
+
+    /// @notice Asserts that the module is enabled for the given Safe.
+    /// @param _safe The Safe address to check.
+    function _assertModuleEnabled(address _safe) internal view {
+        Safe safe = Safe(payable(_safe));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+    }
+
+    /// @notice Internal function to cancel a challenge and emit the appropriate event.
+    /// @param _safe The Safe address for which to cancel the challenge.
+    function _cancelChallenge(address _safe) internal {
+        // Early return if no challenge exists
+        if (challengeStartTime[_safe] == 0) return;
+
+        delete challengeStartTime[_safe];
+        emit ChallengeCancelled(_safe);
+    }
+
+    /// @notice Asserts that the module is not enabled for the given Safe.
+    /// @param _safe The Safe address to check.
+    function _assertModuleNotEnabled(address _safe) internal view {
```

Comment:
If we are fixing nits, this function should move just below `_assertModuleEnabled`.